### ISSUE TITLE
[BUG] Longhorn thinks node is unschedulable

### ISSUE THREAD

**Issue by xorinzor** [longhorn/longhorn#9011]
## Describe the bug

The longhorn manager is spamming my log over and over with this message:
```
time="2024-07-16T11:08:18Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=k8s-worker2 owner=k8s-worker2 replica=pvc-c894115d-a103-4e70-8655-aa5e109c1224-r-6c0ff31b replicaAutoBalanceType=node state=attached volume=pvc-c894115d-a103-4e70-8655-aa5e109c1224
```

But the node seems to be up and running normally, with no indication of it being unschedulable anywhere.
Additionally, some of my volumes are sometimes randomly turning read-only. Deleting and re-creating deployments seems to fix this, but this started happening since a few weeks ago. Unknown if it's related to this problem.

```
kubectl get node
NAME               STATUS   ROLES           AGE    VERSION
k8s-control1       Ready    control-plane   162d   v1.29.5
k8s-worker2        Ready    <none>          162d   v1.29.5
k8s-worker3        Ready    <none>          162d   v1.29.5
```

```
kubectl -n longhorn-system get nodes.longhorn.io
NAME               READY   ALLOWSCHEDULING   SCHEDULABLE   AGE
k8s-worker2        True    true              True          97d
k8s-worker3        True    true              True          97d
```

![image](https://github.com/user-attachments/assets/d7a4b8f0-ad68-4686-85ec-ad031d4df982)

## To Reproduce

Unknown

## Expected behavior

A better error indicating what's actually wrong, with this being correctly reflected in the GUI.
The current error does not appear to be accurate.

## Support bundle for troubleshooting

Sent via email

## Environment

 - Longhorn version: 1.6.2
 - Impacted volume (PV): pvc-c894115d-a103-4e70-8655-aa5e109c1224
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm via ArgoCD
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: kubeadm 1.29.5
   - Number of control plane nodes in the cluster: 1
   - Number of worker nodes in the cluster: 4
 - Node config
   - OS type and version: Raspbian 6.6
   - Kernel version: 6.6.28+rpt-rpi-2712 & 
   - CPU per node: 4 cores
   - Memory per node: 8GB
   - Disk type (e.g. SSD/NVMe/HDD): NVMe
   - Network bandwidth between the nodes (Gbps): 1
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 5

**Comment by ejweber** [comment]
Thanks for the report @xorinzor.

I will follow up more tomorrow, but for now, I think I understand WHY you are seeing the log.

When `replica-auto-balance == best-effort`, the following functions are called frequently:

`getReplicaCountForAutoBalanceNode` -> `listReadySchedulableAndScheduledNodesRO` -> `FilterNodesSchedulableForVolume`

If `FilterNodesSchedulableForVolume` filters out a node for the volume, but there is a replica running on that node, we log the message you are frequently seeing.

https://github.com/longhorn/longhorn-manager/blob/10f6d94b8ebc42f30f90c174469c0f39f4eb9c81/controller/volume_controller.go#L2751-L2755

As you say, all of your nodes are currently running and schedulable. However, `FilterNodesSchedulableForVolume` checks whether there is some disk on a node a replica COULD be scheduled to with `IsSchedulableToDisk`. If an appropriate disk doesn't exist, the node is filtered out (and then readded, as described above).

https://github.com/longhorn/longhorn-manager/blob/10f6d94b8ebc42f30f90c174469c0f39f4eb9c81/scheduler/replica_scheduler.go#L837-L840

Disk `data1` on `k8s-worker2` currently fails the scheduling test because:

`size + info.StorageScheduled` = `161061273600 + 214779822080` = 375841095680 > 372991926272 = `(372991926272 - 0) * (100 / 100) = `(info.StorageMaximum-info.StorageReserved) * (info.OverProvisioningPercentage / 100)`.

In other words, we are logging that there is a replica running on a node we can't schedule a replica to. But the reason we can't schedule a replica to the node is because there is already one running there (taking up space). The log doesn't make sense in this context.

**Comment by xorinzor** [comment]
Thanks for the quick reply!

I'm trying to understand your explanation here:
> In other words, we are logging that there is a replica running on a node we can't schedule a replica to. But the reason we can't schedule a replica to the node is because there is already one running there (taking up space). The log doesn't make sense in this context.

If a replica would be scheduled on the node where there's no space left, wouldn't the UI show one of the replica's as `rebuilding`? From what I can tell in the UI, none of the volumes have any issues with their replica's.

**Comment by ejweber** [comment]
Yeah, I tried to come up with a way of wording it, but it came out confusing no matter what!

We don't need to schedule a replica at all. In the normal reconciliation loop, we check whether replicas need to auto balance between nodes. Part of that check (it's entirely an internal implementation detail) is to determine whether we even COULD schedule a replica to the node in question. For two of your nodes, we couldn't. It doesn't really matter though, as we don't need to rebalance replicas anyway (there are already replicas running on the nodes in question). Unfortunately, we log a warning every time we perform that check, so your logs are flooded.

**Comment by xorinzor** [comment]
aah okay, that clears things up. Thanks!

**Comment by ejweber** [comment]
### Root causes:

1. We log the message because `replica-auto-balance == best-effort` and there is at least one node that doesn't have enough room to schedule a new (unrequired) replica for a volume, but is already running a replica for that volume.
2. We log the message multiple times per reconciliation because the message is generated once per "offending" replica in `listReadySchedulableAndScheduledNodesRO`, and that function runs multiple times every loop.

```
syncVolume
	ReconcileEngineReplicaState
		replenishReplicas (auto balance is enabled)
			getReplenishReplicasCount
				getReplicaCountForAutoBalanceZone (auto balance is least or best effort)
				getReplicaCountForAutoBalanceNode (auto balance is least or best effort)
				getReplicaCountForAutoBalanceNode (auto balance is best effort)
				getReplicaCountForAutoBalanceZone (auto balance is best effort)
```
3. We reconcile twice every five seconds because of https://github.com/longhorn/longhorn-manager/pull/2837, which will be mitigated in `v1.6.3`, `v1.7.0`, etc.

As a result, the user sees 2 (reconciles per size change) x 4 (function calls per reconcile) x 2 (replicas that trigger the message) = 16 messages per 5 second interval.

### Reproduce

1. Install Longhorn `v1.6.2`.
2. Update the setting `replica-auto-balance = best-effort`.
3. Deploy a workload with the following properties:
   - Uses a Longhorn volume that takes up more than half of the available space on each node.
   - Constantly writes to this volume.
4. Observe a flood of log messages.

I used:

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: longhorn-volv-pvc
  namespace: default
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: longhorn
  resources:
    requests:
      storage: 70Gi
---
apiVersion: v1
kind: Pod
metadata:
  name: volume-test
  namespace: default
spec:
  restartPolicy: Always
  containers:
  - name: volume-test
    image: nginx:stable-alpine
    imagePullPolicy: IfNotPresent
    volumeMounts:
    - name: volv
      mountPath: /data
    ports:
    - containerPort: 80
    command: ["sh", "-c", "while true; do head -c 1024 /dev/urandom >> /data/file; sleep 1; done"]
  volumes:
  - name: volv
    persistentVolumeClaim:
      claimName: longhorn-volv-pvc
```

I observed (2 x 4 x 3 = 24 messages / 5 s):

```
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=node state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-004653f7 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-kgxdq owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-8758c8c4 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
[longhorn-manager-bv9lq] time="2024-07-17T16:21:29Z" level=warning msg="Including unschedulable node because the replica is scheduled and running" func="controller.(*VolumeController).listReadySchedulableAndScheduledNodesRO" file="volume_controller.go:2555" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-rw5hf owner=eweber-v126-worker-9c1451b4-kgxdq replica=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc-r-dab794f9 replicaAutoBalanceType=zone state=attached volume=pvc-e89d7311-0eeb-4427-94de-c83c0ffd0bcc
```

**Comment by ejweber** [comment]
Can you help to triage this @derekbit? I mostly just want to confirm we should add it to v1.8.0 with backports to v1.7.x and v1.6.x since the content for v1.7.0 is already set and the severity is low.

**Comment by derekbit** [comment]
> Can you help to triage this @derekbit? I mostly just want to confirm we should add it to v1.8.0 with backports to v1.7.x and v1.6.x since the content for v1.7.0 is already set and the severity is low.

Added. Thank you.

**Comment by c3y1huang** [comment]
Update milestone from v1.9.0 to v1.8.0 because this will be backported to v1.6.3 and v1.7.2, which are planned for release before v1.9.0.

cc @derekbit

**Comment by roger-ryao** [comment]
**Verified on `v1.8.0-dev-20240908` 20240909**

- longhorn `v1.8.0-dev-20240908` https://github.com/longhorn/longhorn/commit/2ce609f871c89ef80bf3decc30b07225bed5288f
- longhorn-manager `v1.8.0-dev-20240908` https://github.com/longhorn/longhorn-manager/commit/e9e0a7f6da6c377ddb9f5f7f57138f4b4a54a000
- longhorn-engine `v1.8.0-dev-20240908` https://github.com/longhorn/longhorn-engine/commit/d3898b14b198e7aae9cd9eb9f1f3579d12acfcf9


**The test steps**
https://github.com/longhorn/longhorn/issues/9011#issuecomment-2233714842

**Result Passed**
- After changing the log level to `Trace`, we can see the related `Including unschedulable node message` in the longhorn-manager pod. This message is not visible with the default log level.
